runtime/lua/vim/shared.lua

```diff
@@ -234,6 +238,26 @@ local function tbl_extend(behavior, deep_extend, ...)
   return ret
 end
 
+--- Merges the values similar to vim.tbl_deep_extend with the **force** behavior,
+--- but the values can be any type, in which case they override the values on the left.
+--- Values will me merged in-place in the first left-most table. If you want the result to be in
+--- a new table, then simply pass an empty table as the first argument `vim.merge({}, ...)`
+function vim.merge(...)
```

Comment:
Is it worth mentioning that this may not be safe to use for dict-style tables where numbers are used as key?

E.g a common pattern is to have the buf number as key - then you could have something like this:

```
vim.merge({ [10] = 'a'}, { [20] = 'b'})
```

Which appears to work, but if you have a `[1] = ...` entry as well, it will override the left value.

```
vim.merge({ [1] = 'a', [10] = 'a'}, { [20] = 'b'})
```

Reply:
Good catch!

This also impacts `tbl_deep_extend`. In there lists always overwrite the left-hand-side, but with my change that would no longer be the case. Will make sure it follows the old behavior.

Right now, `vim.tbl_islist({[10] = 'a'})` returns `true`, but that doesn't seem right? (that was the check that used to be there before)

Not sure what the wanted behavior should be here.

I'm also a bit hesitant to change the behavior of `tbl_islist`.

For `vim.merge(`, I can change it of course.

Maybe we should do the `islist` check there, by checking that all keys are consecutive from 1 to total count? (min key = 1 and max key = count of keys and all keys are numbers)?

This way, your second example would also merge properly. but:
`vim.merge({ [1] = 'a'}, { [20] = 'b'})` would still remove the `1`.

Another approach could be that we merge if one of the params is `can_merge` and the other must be at least a `table`?

That way, merging two list-like tables, will overwrite each other, while your example and my other example above here, would still merge.

Wdyt?

Reply:
I don't think it's a good idea to try being smart about this. I changed the implementation back to how it was for tbl_deep_extend. The same check is now used for `vim.merge`.

We will only merge tables if either the table is empty, or it is not `vim.tbl_islist`.

For your examples, this means:

```lua
vim.merge({ [10] = 'a'}, { [20] = 'b'}) -- result: {[20] = 'b'}

vim.merge({ [1] = 'a', [10] = 'a'}, { [20] = 'b'}) -- result: {[20] = 'b'}
```